types/src/main/scala/aqua/types/Type.scala

```diff
@@ -323,6 +323,17 @@ case class StructType(name: String, fields: NonEmptyMap[String, Type])
     s"$name{${fields.map(_.toString).toNel.toList.map(kv => kv._1 + ": " + kv._2).mkString(", ")}}"
 }
 
+case class StreamMapType(name: String, fields: NonEmptyMap[String, Type])
```

Comment:
Why does `StreamMapType` has fields? How are they used?
I believe this is valid only for particular copy and create use case of this PR.
If we are to expose map type to user, it is better to store key type (string or int) and value type here, like it is for streams